connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecutePlanHolder.scala

```diff
@@ -29,13 +30,20 @@ case class ExecutePlanHolder(
 
   val jobGroupId =
     s"User_${sessionHolder.userId}_Session_${sessionHolder.sessionId}_Request_${operationId}"
+  val events: RequestEvents = RequestEvents(this, new SystemClock())
 
   def interrupt(): Unit = {
     // TODO/WIP: This only interrupts active Spark jobs that are actively running.
     // This would then throw the error from ExecutePlan and terminate it.
     // But if the query is not running a Spark job, but executing code on Spark driver, this
     // would be a noop and the execution will keep running.
     sessionHolder.session.sparkContext.cancelJobGroup(jobGroupId)
+
+    // TODO: The above does not cancel the Connect query and it continues executing.
+    // events.postCanceled()
```

Comment:
Later it could go to https://github.com/apache/spark/pull/41315/files#diff-d5adccdf728da919f25b3d0482f5c2266890ff659b1b436d600ede60be077627R140 (once that PR is finished, WIP so far)